### ISSUE TITLE
pull-local fails when built without libcurl and libsoup

### ISSUE THREAD

**Issue by akiernan** [ostreedev/ostree#1897]
Whilst tidying up my bitbake recipe and testing all combinations of options, if ostree is built without any network transport, I expected `pull-local` to still work, but it fails:

```
$ ostree --repo=repo pull-local ostree-repo
error: This version of ostree was built without libsoup or libcurl, and cannot fetch over HTTP
```

**Comment by cgwalters** [comment]
Yeah, definitely a bug, though it looks a bit ugly to fix since the real `ostree_repo_pull_with_options()` is very entangled with the HTTP-using APIs.

Hm, maybe the simplest way to fix this would be in this case to define an `OstreeFetcher` that only supports `file://`.

**Comment by cmhe** [comment]
I am currently looking into ostree for a project as well, and stumbled over this bug.
